packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -415,7 +436,9 @@ class NestedScrollViewState extends State<NestedScrollView> {
     if (!mounted) {
       return;
     }
-    final bool newHasScrolledBody = _coordinator!.hasScrolledBody;
+    final bool newHasScrolledBody = widget.nestedScrollController
+      ? _nestedScrollController!.hasScrolledBody
+      : _coordinator!.hasScrolledBody;
```

Comment:
There is a lot of duplication between controller and coordinator. Can these not be one API?